File: tmdbClient/app/src/main/java/hootor/com/tmdbclient/model/MovieRepository.java
package hootor.com.tmdbclient.model;

import android.app.Application;
import android.arch.lifecycle.MutableLiveData;

import java.util.ArrayList;
import java.util.List;

import hootor.com.tmdbclient.R;
import hootor.com.tmdbclient.service.MovieDataService;
import hootor.com.tmdbclient.service.RetrofitInstance;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class MovieRepository {

    private ArrayList<Movie> movies = new ArrayList<>();

    private MutableLiveData<List<Movie>> mutableLiveData = new MutableLiveData<>();
    private Application application;

    public MovieRepository(Application application) {
        this.application = application;
    }

    public MutableLiveData<List<Movie>> getMutableLiveData() {

        final MovieDataService movieDataService = RetrofitInstance.getService();

        Call<MovieDBResponse> call = movieDataService.getPopularMovies(application.getApplicationContext().getString(R.string.api_key));
        call.enqueue(new Callback<MovieDBResponse>() {
            @Override
            public void onResponse(Call<MovieDBResponse> call, Response<MovieDBResponse> response) {

                MovieDBResponse movieDBResponse = response.body();

                if (movieDataService != null && movieDBResponse.getResults() != null) {

                    movies = (ArrayList<Movie>) movieDBResponse.getResults();
                    mutableLiveData.setValue(movies);
                }
            }

            @Override
            public void onFailure(Call<MovieDBResponse> call, Throwable t) {

            }
        });
        return mutableLiveData;
    }
}

File: MVVMRetrofitDemo/app/src/main/java/com/android/hootr/mvvmretrofitdemo/model/MovieRepository.kt
package com.android.hootr.mvvmretrofitdemo.model

import android.app.Application
import androidx.lifecycle.MutableLiveData
import com.android.hootr.mvvmretrofitdemo.service.RetrofitInstance
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MovieRepository(val application: Application) {

    private val mutableLiveData: MutableLiveData<List<Result>> = MutableLiveData()

    fun getMutableLiveData(): MutableLiveData<List<Result>> {
        val movieApiService = RetrofitInstance.getService()
        val call = movieApiService.getPopularMovies("e9d6b2db34845e6100bc75f8e904ebc5")

        call?.let {
            it.enqueue(object : Callback<MovieApiResponse?> {

                override fun onFailure(call: Call<MovieApiResponse?>, t: Throwable) {

                }

                override fun onResponse(
                    call: Call<MovieApiResponse?>,
                    response: Response<MovieApiResponse?>
                ) {

                    val movieApiResponse = response.body()

                    movieApiResponse?.let { mar ->
                        mar.results.let {
                            mutableLiveData.value = it
                        }
                    }

                }
            })
        }

        return mutableLiveData
    }
}